### DISCUSSION TITLE
`chainHead_v1_storage` - type: `descendantsHashes` is it worth it?

### DISCUSSION THREAD

**Discussion by josepot** [smol-dot/smoldot#2049] [All]
Hi @tomaka,

I recall a conversation we had a few months ago where you mentioned that for Smoldot to respond to a `descendantsHashes` storage query, the client would still need to download the actual values associated with the storage key. From those values, Smoldot would then compute the hashes. 

If I understood correctly, this was because the P2P protocol doesn’t allow the client to request just the hashes alone, which makes the process somewhat inefficient. As a result, using `descendantsHashes` to check for changes in storage values wasn’t particularly advantageous, and it was generally better to query `descendantsValues` directly instead.

Does this still hold true? Or have there been any changes in this regard since we last discussed it?

**Comment by tomaka** [comment]
That's still true, yes. The issues I still follow in the Polkadot SDK repo haven't gotten any activity.

**Reply by josepot** [reply]
> The issues I still follow in the Polkadot SDK repo

Could you please share the relevant issues for this? 🙏

**Reply by tomaka** [reply]
Actually that would be https://github.com/polkadot-fellows/RFCs/pull/9

The RFC tried to redesign the entire light client protocol, but if we want to simply make full nodes return hashes (and keep the flaws in the existing protocol) then the changes would be very small

**Reply by josepot** [reply]
> then the changes would be very small

Could you please show us the ropes? 🙏 We (the polkadot-api team) would like to give it a shot.

**Reply by tomaka** [reply]
It's a modification of the protocol, so you need an RFC. All you need to modify is add a flag that asks the full node for only hashes instead of hashes plus values.

**Reply by tomaka** [reply]
The modifications to do in Substrate are out of my reach as the code is very difficult to understand, but the modifications to the protocol are very small, and someone who knows how it works (Emeric, Arkadiy, maybe others, I don't know) can probably do the changes in Substrate very easily.
